### ISSUE TITLE
[Modal] overlay fullscreen modal action height does not grow

### ISSUE THREAD

**Issue by keshmir** [fomantic/Fomantic-UI#692]
# Bug Report
The action area of fullscreen overlay modal allocates height only for one row of buttons.  If there is content in action area that requires more height it gets cut off.  This works ok with the original fullscreen (without overlay).

## Steps to reproduce
1. Run this fiddle: https://jsfiddle.net/9sy2rxuq/19/
2. Click on Fullscreen version 1.  This button launches full screen overlay with lots of buttons in the action area. (resize your screen if necessary to get two rows of buttons).
3) Dismiss and click on version 2. This version is the original fullscreen without overlay and the action area has height to accommodate second row of buttons.

## Expected result
The action area should adjust its height to show both rows of buttons

## Actual result
The second row of buttons gets cut off

## Testcase
https://jsfiddle.net/9sy2rxuq/19/

## Version
2.7.4

**Comment by keshmir** [comment]
So I played around with this and the issue is actually with the way overlay modal is implemented.  This line:

  min-height: calc(100vh - 9.1rem);

Makes the hard coded assumption of the height of the action area.  So in order to get a separate height for my actions, I can just play with this until I get what I need for my purpose.  However, an ideal solution would be to make action grow as needed like the current fullscreen (without overlay).  Any pointers on how to approach that?

**Comment by lubber-de** [comment]
Thanks for investigation. In fact the normal (non overlay) fullscreen behaves the same whenever the content of the action (btw this applies for the header as well) grows in height: The overall modal increases in height. The normal fullscreen indeed does respect this a bit more because there is more room left around the modal, but if you exhaust this by adding unusual tons of buttons, so the action div grows even more, you will gain the same effect (modal moves out of viewport) 
The calc logic is similar to the normal fullscreen variant. 
What you desire is, that the content div should shrink whenever the action div grows to keep the overall dimension the same all the time.
I played around a bit but had no satisfying success.
The only 'solution' which came to my mind is an additional class which definately doubles the action row size by, as, you already figured out, changing the calc values.

**Comment by keshmir** [comment]
makes perfect sense.  I think I like the idea of doubling (and perhaps tripling if needed) as additional classes.  On what element should this class be applied?  at the div.action level?

**Comment by lubber-de** [comment]
That's the problem here: The action as well as the header div don't have and height setting. Only the content has. By the current logic the additional class should be set to the whole modal, so the css selector is more flexible.
Anyhow this overall feels wrong. At least if thinking about implementing such "hack" into the core.
I still believe we need to find a solution to automatically shrink the content whenever header or action grow in size on a overlay fullscreen modal.
So, as you already suggested in your jsfiddle, as a temporary workraround you could use something like

```css
  .ui.doubleaction.overlay.fullscreen.modal > .content.content.content {
      min-height: calc(100vh - 8.1rem);
  }
```
But I wouldn't implement such class into the core as said.

**Comment by keshmir** [comment]
I completely agree that this solution is hacky and not appropriate to be built into core.  For now I just want to migrate my modals to overlay and this was stopping me.
